src/w3d/lib/gcd_lcm.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Greatest Common Divisor/Lowest Common Multiple
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "gcd_lcm.h"
+
+unsigned int Greatest_Common_Divisor(unsigned int a, unsigned int b)
+{
+    unsigned int c = b;
+    unsigned int d = a;
+
+    if (b) {
+        int e;
+
+        do {
+            e = d % c;
+            d = c;
+            c = e;
+        } while (e);
+    }
+
+    return d;
+}
+
+unsigned int Least_Common_Multiple(unsigned int a, unsigned int b)
+{
+    if (b) {
+        return b * a / Greatest_Common_Divisor(b, a % b);
+    } else {
+        return 0 / a;
```

Comment:
This looks wrong.

Reply:
I took another look at this in a few different binaries and re-did it.